### ISSUE TITLE
Research errors in consensus integration test

### ISSUE THREAD

**Issue by GalRogozinski** [iotaledger/goshimmer#1847]
## Description

In #1846 a conflict spam integration test was introduced that spam conflicts sequentially with a delay of 1 second.
In case that delay is removed the following errors may occur:
1. `tips could not be selected: not able to make sure that all inputs are in the past cone of selected tips` 
2. `bad request: not all consumedOutputs of transaction are solid: transaction not solid`
3. `bad request: Failed to issue transaction TransactionID(FRKcecJcMTqYd7LqAhsNnPUrstJdxq2CkVbuwtSzuo2Y): like references could not be prepared: branchID can't be retrieved: failed to retrieve StructureDetails of MessageID`
4. `Different GOFs on txs between peers`

We need to understand the underlying issue(s)

**Comment by GalRogozinski** [comment]
Besides the above errors there is also an error I saw in the simple double spend test inroduced in #1764.
Seems that nodes didn't reach the same GoF

It seems to happen very rarely though...
https://github.com/iotaledger/goshimmer/runs/4292533255?check_suite_focus=true#step:5:573
```
tester    |     testutil.go:573: 
tester    |         	Error Trace:	testutil.go:573
tester    |         	            				consensus_test.go:120
tester    |         	Error:      	Condition never satisfied
tester    |         	Test:       	TestSimpleDoubleSpend
```

**Comment by jonastheis** [comment]
This should be fixed now since #1846 is merged into #1770.
